crates/bevy_reflect/src/lib.rs

```diff
@@ -1213,62 +1222,57 @@ mod tests {
-            "my_alias::DerivePathG<bevy_reflect::tests::Param, 10>"
-        );
-        assert_eq!(
-            DerivePathNameG::<Param>::type_path(),
-            "my_alias::MyDerivePathNameG<bevy_reflect::tests::Param>"
-        );
+        assert_type_paths! {
+            Derive => "bevy_reflect::tests::Derive", "Derive",
+            DerivePath => "my_alias::DerivePath", "DerivePath",
+            DerivePathName => "my_alias::MyDerivePathName", "MyDerivePathName",
+            DeriveG<Param> => "bevy_reflect::tests::DeriveG<bevy_reflect::tests::Param>", "DeriveG<Param>",
+            DerivePathG<Param, 10> => "my_alias::DerivePathG<bevy_reflect::tests::Param, 10>", "DerivePathG<Param, 10>",
+            DerivePathNameG<Param> => "my_alias::MyDerivePathNameG<bevy_reflect::tests::Param>", "MyDerivePathNameG<Param>",
+            Macro => "my_alias::Macro", "Macro",
+            MacroName => "my_alias::MyMacroName", "MyMacroName",
+            MacroG<Param, 10> => "my_alias::MacroG<bevy_reflect::tests::Param, 10>", "MacroG<Param, 10>",
+            MacroNameG<Param> => "my_alias::MyMacroNameG<bevy_reflect::tests::Param>", "MyMacroNameG<Param>",
+        }
+    }
 
-        assert_eq!(Macro::type_path(), "my_alias::Macro");
-        assert_eq!(MacroName::type_path(), "my_alias::MyMacroName");
-        assert_eq!(
-            MacroG::<Param, 10>::type_path(),
-            "my_alias::MacroG<bevy_reflect::tests::Param, 10>"
-        );
-        assert_eq!(
-            MacroNameG::<Param>::type_path(),
-            "my_alias::MyMacroNameG<bevy_reflect::tests::Param>"
-        );
+    #[test]
+    fn std_type_paths() {
+        #[derive(Clone)]
+        struct Type;
 
-        assert_eq!(Derive::short_type_path(), "Derive");
-        assert_eq!(DerivePath::short_type_path(), "DerivePath");
-        assert_eq!(DerivePathName::short_type_path(), "MyDerivePathName");
+        impl TypePath for Type {
+            fn type_path() -> &'static str {
+                // for brevity in tests
+                "Long"
+            }
 
-        assert_eq!(DeriveG::<Param>::short_type_path(), "DeriveG<Param>");
-        assert_eq!(
-            DerivePathG::<Param, 10>::short_type_path(),
-            "DerivePathG<Param, 10>"
-        );
-        assert_eq!(
-            DerivePathNameG::<Param>::short_type_path(),
-            "MyDerivePathNameG<Param>"
-        );
+            fn short_type_path() -> &'static str {
+                "Short"
+            }
+        }
 
-        assert_eq!(Macro::short_type_path(), "Macro");
-        assert_eq!(MacroName::short_type_path(), "MyMacroName");
-        assert_eq!(MacroG::<Param, 10>::short_type_path(), "MacroG<Param, 10>");
-        assert_eq!(
-            MacroNameG::<Param>::short_type_path(),
-            "MyMacroNameG<Param>"
-        );
+        assert_type_paths! {
+            u8 => "u8", "u8",
+            Type => "Long", "Short",
+            &Type => "&Long", "&Short",
```

Comment:
Just to clarify, this wouldn't work for `&&Type` since `&'static Type` doesn't implement `Reflect`, right?